raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -1688,14 +1694,39 @@ private long pollCurrentState(long currentTimeMs) throws IOException {
         }
     }
 
+    private void pollListeners() {
+        // Register any listeners added since the last poll
+        while (!pendingListeners.isEmpty()) {
```

Comment:
I doubt we would use it in practice, though I guess it would open the door to changing roles dynamically, which might be interesting in the future. That said, it was simple to add and useful in testing since it gave me an easy way to initialize a state where a listener had not caught up.